### DISCUSSION TITLE
tutorials/ui/custom_gui_controls

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#234] [User-contributed notes]
# tutorials/ui/custom_gui_controls

So many controls...: Yet there are never enough. Creating your own custom controls that act just the way you want them to is an obsession of almost every GUI programmer. Godot provides plenty of th...

https://docs.godotengine.org/en/stable/tutorials/ui/custom_gui_controls.html

**Comment by UnrealSoftware** [comment]
The  C# sample code for Notifications doesn't compile. The parameter is "int what" but all the values in  the cases e.g. NotificationMouseEnter are of type long. Can be fixed easily by replace "switch (what)" with "switch ((long)what)"
